samples/nrf9160/lwm2m_client/src/main.c

```diff
@@ -108,6 +109,22 @@ static int query_modem(const char *cmd, char *buf, size_t buf_len)
 	return 0;
 }
 
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_SIGNAL_MEAS_INFO_OBJ_SUPPORT)
+void ncell_meas_work_handler(struct k_work *work)
+{
+	lte_lc_neighbor_cell_measurement(LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT);
+}
+
+K_WORK_DEFINE(ncell_meas_work, ncell_meas_work_handler);
+
+void ncell_meas_work_timer_handler(struct k_timer *dummy)
+{
+	k_work_submit(&ncell_meas_work);
+}
+
+K_TIMER_DEFINE(ncell_meas_work_timer, ncell_meas_work_timer_handler, NULL);
```

Comment:
Instead of a timer/work combo I suggest simply using delayable work (https://docs.zephyrproject.org/latest/reference/kernel/threads/workqueue.html#delayable-work) - a work that is executed after a specific time.